nnvm/tests/python/frontend/onnx/test_forward.py

```diff
@@ -380,6 +380,94 @@ def test_lrn():
     verify_lrn((5, 5, 5, 5), 3, 'float32')
     verify_lrn((5, 5, 5, 5), 3, 'float32', alpha=0.0002, beta=0.5, bias=2.0)
 
+def _test_upsample_nearest():
+    scale = 2
+    in_shape = (1, 1, 3, 3)
+    out_shape = (1, 1, 3 * scale, 3 * scale)
+    scale = float(scale)
```

Comment:
Are you sure the result of float(scale) is float32? The "Cast" warning you got might be due to this.